src/domain-services/flows/flow-service.ts

```diff
@@ -473,91 +485,88 @@ export class FlowService {
+    // 4. Query parent flows progressively in chunks
     const parentFlows = await this.progresiveSearch(
       models,
       parentFlowsRef,
       1000,
       0,
-      false, // Stop on batch size
+      false, // Do not stop on batch size
       [],
       { activeStatus: true }
     );
 
+    // 5. Retrieve flow objects using the flow object filters
     const flowObjectsWhere =
       buildWhereConditionsForFlowObjectFilters(flowObjectFilters);
-
     const flowObjects = await this.flowObjectService.getFlowFromFlowObjects(
       models,
       flowObjectsWhere
     );
 
-    // Once we get the flowObjects - we need to keep only those that are present in both lists
-    const filteredParentFlows = parentFlows.filter((parentFlow) =>
-      flowObjects.some(
-        (flowObject) =>
-          flowObject.id === parentFlow.id &&
-          flowObject.versionID === parentFlow.versionID
+    // 6. Build a Set for flowObjects for fast lookup\n  (using a composite key of id and versionID)\n
+    const flowObjectsSet = new Set(
+      flowObjects.map(
+        (flowObject) => `${flowObject.id}|${flowObject.versionID}`
       )
     );
 
-    // Once we have the ParentFlows whose status are 'parked'
-    // We keep look for the flowLinks of those flows to obtain the child flows
-    // that are linked to them
-    const childFlowsIDs: FlowId[] = [];
+    // 7. Filter parent flows that are present in the flowObjects list
+    const filteredParentFlows = parentFlows.filter((parentFlow) => {
+      const key = `${parentFlow.id}|${parentFlow.versionID}`;
+      return flowObjectsSet.has(key);
+    });
+
+    // 8. Build a Set of filtered parent flow IDs for quick membership checking
+    const filteredParentFlowIds = new Set(
+      filteredParentFlows.map((flow) => flow.id)
+    );
+
+    // 9. Extract child flow IDs from flowLinks where the parent is in the filtered set
+    const childFlowsIDsSet = new Set<FlowId>();
     for (const flowLink of flowLinks) {
-      if (
-        filteredParentFlows.some(
-          (parentFlow) => parentFlow.id === flowLink.parentID
-        )
-      ) {
-        childFlowsIDs.push(flowLink.childID);
+      if (filteredParentFlowIds.has(flowLink.parentID)) {
+        childFlowsIDsSet.add(createBrandedValue(flowLink.childID));
       }
     }
 
+    // 10. Retrieve child flows
     const childFlows = await models.flow.find({
       where: {
         deletedAt: null,
         activeStatus: true,
-        id: {
-          [Op.IN]: childFlowsIDs,
-        },
+        id: { [Op.IN]: [...childFlowsIDsSet] },
       },
       distinct: ['id', 'versionID'],
     });
 
-    // Once we have the child flows, we need to filter them
-    // using the flowObjectFilters
-    // This search needs to be also done by chunks
-    return childFlows.map((ref) => ({
+    // 11. Map child flows to UniqueFlowEntity and return the result
+    const result = childFlows.map((ref) => ({
       id: createBrandedValue(ref.id),
```

Comment:
There is no need for using `createBrandedValue()` here
```suggestion
      id: ref.id,
```